Tella/Scenes/Settings/Views/Servers/GDrive/SelectSharedDrive.swift

```diff
@@ -0,0 +1,68 @@
+//
+//  SelectSharedDrive.swift
+//  Tella
+//
+//  Created by gus valbuena on 5/20/24.
+//  Copyright © 2024 HORIZONTAL. All rights reserved.
+//
+
+import SwiftUI
+import GoogleAPIClientForREST
+
+struct SelectSharedDrive: View {
+    @State var selectedDrive: String = ""
+    @State var sharedDrives: [GTLRDrive_Drive] = []
+    var body: some View {
+        ContainerView {
+            VStack(alignment: .leading){
+                ForEach(sharedDrives, id: \.identifier) { drive in
+                    DriveCardView(sharedDrive: drive, selectedDrive: $selectedDrive)
+                }
+
+                Spacer()
+            }
+            .padding(.vertical, 8)
+            .background(Color.white.opacity(0.03))
+            .toolbar {
+                LeadingTitleToolbar(title: "Select shared drive")
+                // remove this after merging with uwazi relationships
+                ToolbarItem(placement: .navigationBarTrailing) {
+                    Button(action: {
+                        navigateTo(destination: SuccessLoginView(navigateToAction: {}, type: .gDrive))
+                    }) {
+                        Image("report.select-files")
+                            .resizable()
+                            .frame(width: 24, height: 24)
+                    }
+                }
+            }
+        }
+    }
+}
+
+struct DriveCardView: View {
+    var sharedDrive: GTLRDrive_Drive
+    @Binding var selectedDrive: String
+    var body: some View {
+        Button(action: {
+            self.selectedDrive = sharedDrive.identifier ?? ""
```

Comment:
What do you think if we add the selectedDrive action to the view model ?

Reply:
solved in this [commit](https://github.com/Horizontal-org/Tella-iOS/pull/147/commits/485fb87adfed525f40f9a290a24d501ed0e19d50) ✅